Adjust the code generated for enum types to avoid conflicting with names not reserved by the Flatbuffers IDL

insertjokehere opened the pull request:
Fixes #8052

The Flatbuffers IDL allows for enums to have values called `value`, `values`, `minValue` or `maxValue`, but if these names are used for values in an enum, the generated Dart code won't compile.

This modifies the Dart generator to use `$value`, `$values`, `$minValue` and `$maxValue` instead, so that valid Flatbuffer schemas can be used with Dart

vaind reviewed (commented):
Thanks for the contribution.

This is quite a breaking change - it would break anyone accessing these properties. Instead, I'd suggest changing the field name in case it conflicts, which would not break anyone because they couldn't have used that name anyway. That is what we do for conflicts with reserved keywords too (see uses of `namer_`).

insertjokehere commented:
Sorry for the delay getting back to this - it slipped way down my list for a bit.

Your criticism about the degree to which this was a breaking change is valid - I've reworked this to set up a second `Namer` instance with an extended list of "keywords" that it needs to escape. C++ isn't a language I use very much, so I'm not sure if the way I'm doing this is correct/idiomatic, but it Works For Me. I've also added a test to ensure that we generate valid code for enums that use the reserved words as values.

Thanks.
